### PR TITLE
User Interface Branch _ MILESTONE 2

### PR DESCRIPTION
**Pull request by anapdh**
This branch features:

- Implement info we need from the player (the position);
- Implement what we need to display for the player (board and chosen position on board);
- Display prompts for users (hint: puts/print), collect user input (hint: get.chomp), and only mock the basic flow of the game;
- Game is executable.

### CODE REVIEW

File: bin/main.rb [hunk 1]
```diff
@@ -1,3 +1,123 @@
...
+    play(arr)
+  end
+end
+
+def display(board)
+  puts "      #{board[0]} | #{board[1]} | #{board[2]}
+      ----------
+      #{board[3]} | #{board[4]} | #{board[5]}
+      ----------
+      #{board[6]} | #{board[7]} | #{board[8]}"
```
Comment (voscarmv) [inline bin/main.rb]: This can be an attribute of a board object created from a Board class.

File: bin/main.rb [hunk 2]
```diff
@@ -1,3 +1,123 @@
...
+    winning = true
+    puts "Congratulations #{player1.name}, you won!"
+    break
+  elsif board[1] == board[4] && board[4] == board[7] && board[1] != ' '
+    winning = true
+    puts "Congratulations #{player1.name}, you won!"
+    break
+  elsif board[2] == board[5] && board[5] == board[8] && board[2] != ' '
+    winning = true
+    puts "Congratulations #{player1.name}, you won!"
```
Comment (voscarmv) [inline bin/main.rb]: It's not required to do victory checks in this stage of the project, but this could also go in the `Board` class mentioned earlier.

**Review by voscarmv — CHANGES_REQUESTED** [review]
# Nice work so far, team. 👍 

- You weren't supposed to implement the game logic in this step yet, but kudos for getting ahead and coding it in any way!

## However, there are some changes required 👀 🧰 🔨 🔧 

- Please use `require_relative` instead of `require` for including the `player.rb` file in your `main`.
- Please ask the players for their names at the beginning instead of hardcoding them yourself.
- Currently, your game does not check for overlapping moves. Before placing a player's move on the board, please make sure that there is not a mark in the same beforehand.

**[OPTIONAL]**

- Let's try to better modulate our code. This is not a requirement for this step of the project, but it'll help you in the next stage: A lot of the code in your `main` file can be separated into different `lib` files. For example, all the sections related to the board, such as printing the board, or checking if a player won, can be placed inside a `board` class in `lib`. Please take some time to modulate your code in this way.

### Submit these changes and request a new code review! 😄 👍

**Review by javitocor — CHANGES_REQUESTED** [review]
# STATUS CHANGES REQUIRED ♻️ 

Hello team! ✋ 
A great job so far in this milestone! 👏 
Let's just implement some basic requirements and ready to go! 🚀 
Kindly check my comments below 👇  apply the changes and resubmit your project for a new code review ⏩ 
Good luck with it! 😄 
## Mandatory changes
- In the board, show the player the available positions, ie display 0,1,2,3,4.... in the board so the player knows the spot he/she wants to play
- No game logic (for example checking who is a winner) is required in this milestone - instead, you can just display "The winner is X" no matter who won. You can use a while loop with a boolean variable to display such message. ie.
`
```game_on = true

# initial instructions for player

while game_on
  # loop for each move

  if winner # and/or draw (the exact condition in this milestone is not important)
    game_on = false
  end
end
````
Again, you do not need to compute who won but create a loop(recommended is `while` but anything else will do) for any turn

**Review by teekaytech — CHANGES_REQUESTED** [review]
# Status: Approved :green_circle: 

### Congratulations team, this milestone has passed all the mandatory requirements. It is therefore approved :heavy_check_mark:

You can now merge and checkout to the next milestone :tada: 
Good luck!

* * * 
**Taofeek Olalere
[Twitter](https://twitter.com/ola_lere) | [Github](https://github.com/teekaytech) | @taofeekolalere on Slack**
__Happy Coding :smiley:__

**Pull request merged**